### ISSUE TITLE
Console command doctrine:generate:entities crates whitespaces.

### ISSUE THREAD

**Issue by codeName007** [doctrine/DoctrineBundle#432]
Steps to reproduce bug:

1) Installing the Symfony Installer:
````
sudo curl -LsS http://symfony.com/installer -o /usr/local/bin/symfony
sudo chmod a+x /usr/local/bin/symfony
````
2) Creating the Symfony Application:
````
symfony new bug_justification
````
3) Create product entity:
````
cd bug_justification/
mkdir src/AppBundle/Entity
touch src/AppBundle/Entity/Product.php
````
4) Add next code to entity:
````php
<?php

namespace AppBundle\Entity;
use Doctrine\ORM\Mapping as ORM;

/**
* @ORM\Entity(repositoryClass="AppBundle\Entity\ProductRepository")
* @ORM\Table(name="product")
*/
class Product
{
    /**
    * @ORM\Column(type="integer")
    * @ORM\Id
    * @ORM\GeneratedValue(strategy="AUTO")
    */
    protected $id;
    /**
    * @ORM\Column(type="string", length=100)
    */
    protected $name;
    /**
    * @ORM\Column(type="decimal", scale=2)
    */
    protected $price;
    /**
    * @ORM\Column(type="text")
    */
    protected $description;
}
````
5) Run console command:
````
php app/console doctrine:generate:entities AppBundle/Entity/Product
````
As result doctrine update entity file with additional code, and add methods like: getName, getPrice etc and this methods will have own phpDoc with whitespaces. And when you add this entity to git and try to check it with: `git diff --cached --check src/AppBundle/Entity/Product.php` you will receive something like:

![001](https://cloud.githubusercontent.com/assets/5052324/7899635/ba71bf3e-0733-11e5-95cd-7946b4007423.png)

**Comment by phansys** [comment]
AFAIK, it uses [EntityGenerator::$getMethodTemplate](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Tools/EntityGenerator.php#L229) from ```doctrine/orm```, and it seems just fine. Could you locate this class in your setup and check how it looks?

**Comment by codeName007** [comment]
I found what exactly generates this error.
In file: `./vendor/doctrine/orm/lib/Doctrine/ORM/Tools/EntityGenerator.php`
at method: `generateEntityStubMethod`
present next code: `$variableType   = $typeHint ? $this->getType($typeHint) . ' ' : null;`.
if replace it to: `$variableType   = $typeHint ? $this->getType($typeHint) : null;` - everything will be fine. 
And i decided that i should fix it at docrtrine repo, but it have already fixed. And it mean that we just need to update `doctrine/orm` version at `composer.json`. I update `composer.json` at my project `bug_justification` then run composer update and then run console command again - and all become good. And now i have no idea where i should update this version of doctrine to solve this problem...

Can you advise something, and what you think about it?

**Comment by phansys** [comment]
Since this bugfix is merged in```doctrine/orm: 2.5``` (doctrine/doctrine2@49bca517) and [latest symfony-standard (2.7.0)](https://github.com/symfony/symfony-standard/blob/v2.7.0/composer.json#L12) has a constraint for ```doctrine/orm: <2.5```, you should wait until symfony updates the constraint or update manually the constraint in your project under your own risk.

**Comment by codeName007** [comment]
Yes of course.
But, is it possible to help symfony to update version of `doctrine/orm` at `composer.json` to something like: `"doctrine/orm": "~2.5,<2.6",` ? (i checked it, it works).
Moreover lately [doctrine](https://github.com/doctrine/doctrine2) declared that their version 2.5 has passing build (if I understood everything correctly).

**Comment by phansys** [comment]
That is not possible because ```symfony/framework-standard-edition: 2.7.0``` [keeps compatibility with PHP 5.3](https://github.com/symfony/symfony-standard/blob/v2.7.0/composer.json#L10) while ```doctrine/orm: 2.5``` [has dropped it](https://github.com/doctrine/doctrine2/blob/v2.5.0/composer.json#L16).

**Comment by Ocramius** [comment]
It is actually possible, you'll just 2.4.x if you're running on 5.3.x

**Comment by phansys** [comment]
But doctrine/doctrine2@49bca517 isn't merged in [2.4](https://github.com/doctrine/doctrine2/blob/2.4/lib/Doctrine/ORM/Tools/EntityGenerator.php#L1109), or am I missing something?

**Comment by Ocramius** [comment]
It's only in 2.5, yes. It's not a bugfix worth backporting

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Comment by codeName007** [comment]
Ok, i understand.
It's so pity for me that i can't do something helpful in this direction...
Tell me please, what i should do with this issue?

**Comment by phansys** [comment]
IMO, since the bug is fixed in 2.5 and it won't be backported, you should close this issue. Also, you could update your requirement for ```doctrine/orm``` in your project under your own risk, and leave a comment here about how your "fix" works in order to help other people experiencing the same situation.

**Comment by codeName007** [comment]
Ok. Thank you very much!
